corehq/util/datadog/utils.py

```diff
@@ -100,3 +100,50 @@ def bucket_value(value, buckets, unit=''):
         if value < bucket:
             return "lt_{:03}{}".format(bucket, unit)
     return "over_{:03}{}".format(buckets[-1], unit)
+
+
+def maybe_add_domain_tag(domain_name, tags):
+    """Conditionally add a domain tag to the given list of tags"""
+    if (settings.SERVER_ENVIRONMENT, domain_name) in settings.DATADOG_DOMAINS:
+        tags.append('domain:{}'.format(domain_name))
+
+
+def load_counter(load_type, source, domain_name, extra_tags=None):
+    """Make a function to track load by counting touched items
+
+    :param load_type: Load type (`"case"`, `"form"`, `"sms"`). Use one
+    of the convenience functions below (e.g., `case_load_counter`)
+    rather than passing a string literal.
+    :param source: Load source string. Example: `"form_submission"`.
+    :param domain_name: Domain name string.
+    :param extra_tags: Optional list of extra datadog tags.
+    :returns: Function that adds load when called: `add_load(value=1)`.
+    """
+    from corehq.util.datadog.gauges import datadog_counter
+    tags = ["src:%s" % source]
+    if extra_tags:
+        tags.extend(extra_tags)
+    maybe_add_domain_tag(domain_name, tags)
+    metric = "commcare.load.%s" % load_type
```

Comment:
Someone's given me this feedback before I think also for datadog metrics and I found it thoughtful, so I'll pass it on. In order to make this pass the grep test, you could ask the caller function to pass in the full metric name (e.g. `"commcare.load.form"`), and then just _assert_ that it starts with `commcare.load.`. That's maybe just one way to do it, but the overall goal is that if someone sees the metric in datadog, they can just search for "commcare.load.form" and find it directly without having to reason about code they might not understand so well.

Reply:
I think in this case they can grep for `load_counter` to get the same thing. The thing I don't like about listing out the whole thing every time is it adds a fair amount of duplication and makes the lines much longer (or wrapped to multiple lines), which makes the code more verbose and harder to read.

Reply:
Added grep notes in e402302d48b1f8e4c2cd1b7c7287b24b052e7101